### PR TITLE
Promote from BLAIS5-3841 to main

### PR DESCRIPTION
**Pull request by social-surveys-blaise-concourse**
Auto generated by concourse:

Promote from BLAIS5-3841 to main

### CODE REVIEW

File: authenticate/claims.go [hunk 1]
```diff
@@ -19,7 +20,14 @@ func (uacClaims *UACClaims) AuthenticatedForInstrument(instrumentName string) bo
 	if strings.EqualFold(uacClaims.UacInfo.InstrumentName, instrumentName) {
 		return true
 	}
-	if uacClaims.UacInfo.InstrumentName == "dia2299a" && instrumentName == "dia2299b" {
+	return uacClaims.CheckDiaInstrument(uacClaims.UacInfo.InstrumentName, instrumentName)
+}
+
+func (uacClaims *UACClaims) CheckDiaInstrument(instrumentName1, instrumentName2 string) bool {
+	diaA, _ := regexp.MatchString(`^dia\d{4}a$`, instrumentName1)
```
Comment (JAWilliamsONS) [inline authenticate/claims.go:27]: I'm not a massive fan of using the blank identifier as Go is supposed to force you in to handling your errors. I would write something like:

	diaA, err := regexp.MatchString(`^dia\d{4}a$`, instrumentName1)
	if err != nil {
		fmt.Print(err)
	}
	diaB, err := regexp.MatchString(`^dia\d{4}b$`, instrumentName2)
	if err != nil {
		fmt.Print(err)
	}
	if diaA && diaB {
		return true
	}
	return false
}

So err if handled when it is thrown

File: authenticate/claims_test.go [hunk 2]
```diff
@@ -25,13 +25,27 @@ var _ = Describe("Claims", func() {
 		}
 	)
 
+	DescribeTable("CheckDiaInstrument",
+		func(instrumentName1, instrumentName2 string, expected bool) {
+			// Use the checkDiaInstrument function from the imported package
+			Expect(claim.CheckDiaInstrument(instrumentName1, instrumentName2)).To(Equal(expected))
+		},
+		Entry("both diaA and diaB", "dia1234a", "dia5678b", true),
```
Comment (JAWilliamsONS) [inline authenticate/claims_test.go:33]: Are we sure that this scenario is correct? Will users actually be going from a questionnaire 123 to 456? In the ticket it just says 

The questionnaires will always start with dia.
The primary questionnaire will be suffixed with a a
The secondary questionnaire will be suffixed with a b 

Should this actually return false?

**Review by JAWilliamsONS — COMMENTED** [review]
left some comments for you 👍

File: authenticate/claims_test.go [hunk 3]
```diff
@@ -25,13 +25,27 @@ var _ = Describe("Claims", func() {
...
+		Entry("only diaA", "dia1234a", "notdia5678b", false),
+		Entry("only diaB", "notdia1234a", "dia5678b", false),
+		Entry("neither diaA nor diaB", "notdia1234a", "notdia5678b", false),
+		Entry("different case for diaA", "Dia1234a", "dia5678b", false),
+		Entry("different case for diaB", "dia1234a", "DIA5678B", false),
+		Entry("both in different case", "DIA1234A", "DIA5678B", false),
+		Entry("invalid names", "bacon", "ham", false),
+	)
+
```
Comment (JAWilliamsONS) [inline authenticate/claims_test.go:42]: Will need to be reversed if you implement the above logic change

File: authenticate/claims.go (hunk 1 above)
Comment (alistair-platt) [reply authenticate/claims.go]: Chatted this through with James, and added the error code, but let the logic the same

File: authenticate/claims_test.go (hunk 2 above)
Comment (alistair-platt) [reply authenticate/claims_test.go]: This was a good spot from James, and is now changed

**Review by JAWilliamsONS — APPROVED** [review]

**Pull request merged**